scripts/core/editor3/components/Editor3.jsx

```diff
@@ -133,8 +144,19 @@ export class Editor3Component extends React.Component {
         if (this.allowItem(e)) {
             // Firefox ignores the result of onDragOver and accept the item in all cases
             // Here will be tested again if the item is allowed
-            this.props.dragDrop(e);
+
+            if (this.isFileUpload(e)) {
+                const files = e.originalEvent.dataTransfer.files;
+
+                this.props.insertImages(files);
+            } else {
+                this.props.dragDrop(e);
```

Comment:
I think the Redux friendly way here is to leave the original implementation, and instead do this logic inside the action, like this:
```diff
diff --git a/scripts/core/editor3/actions/editor3.jsx b/scripts/core/editor3/actions/editor3.jsx
index a8b540e8..d9e20267 100644
--- a/scripts/core/editor3/actions/editor3.jsx
+++ b/scripts/core/editor3/actions/editor3.jsx
@@ -46,6 +46,12 @@ export function handleEditorTab(e) {
  * @description Creates the editor drop action.
  */
 export function dragDrop(e) {
+    const transfer = e.originalEvent.dataTransfer;
+
+    if (transfer.types[0] === 'Files') {
+        return insertImages(transfer.files);
+    }
+
     return {
         type: 'EDITOR_DRAG_DROP',
         payload: e
```